server/eclipse-project/src/main/java/us/freeandfair/corla/asm/AbstractStateMachine.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Free & Fair Colorado RLA System
+ * 
+ * @title ColoradoRLA
+ * @created Aug 8, 2017
+ * @copyright 2017 Free & Fair
+ * @license GNU General Public License 3.0
+ * @author Joe Kiniry <kiniry@freeandfair.us>
+ * @description A system to assist in conducting statewide risk-limiting audits.
+ */
+
+package us.freeandfair.corla.asm;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import us.freeandfair.corla.Main;
+
+/**
+ * @description A generic Abstract State Machine (ASM).
+ * @trace asm.asm
+ * @author Joseph R. Kiniry <kiniry@freeandfair.us>
+ * @author Daniel M. Zimmerman <dmz@freeandfair.us>
+ * @version 0.0.1
+ */
+public abstract class AbstractStateMachine implements Serializable {
+  /**
+   * The serialVersionUID.
+   */
+  private static final long serialVersionUID = 1; 
+  
+  /**
+   * This ASM's set of states.
+   */
+  protected final Set<ASMState> my_states;
+  
+  /**
+   * This ASM's initial state.
+   */
+  protected final ASMState my_initial_state;
+  
+  /**
+   * This AMS's final states.
+   */
+  protected final Set<ASMState> my_final_states;
+  
+  /**
+   * This ASM's set of events.
+   */
+  protected final Set<ASMEvent> my_events;
+  
+  /**
+   * A map from (state, event) pairs to state.
+   */
+  protected final Set<ASMTransition> my_transition_function; 
+  
+  /**
+   * The relation between UI events and ASM transitions.
+   */
+  protected final UIToASMEventRelation my_ui_to_asm_relation = 
+      new UIToASMEventRelation();
+  
+  /**
+   * The current state of this ASM. Initialized to the initial state provided
+   * in the constructor.
+   */
+  protected ASMState my_current_state;
+  
+  /**
+   * Constructs an ASM. This constructor takes ownership of all the 
+   * Collections passed to it.
+   * 
+   * @param the_states the states of the new ASM.
+   * @param the_events the events of the new ASM.
+   * @param the_transition_function the transition function of the new ASM. This
+   * function, represented as a set of ASMTransitionFunction elements, need only
+   * specify legal transitions; all unspecified transitions are considered illegal.
+   * @param the_initial_state The initial state of the new ASM.
+   * @param the_final_states The final states of the new ASM.
+   */
+  public AbstractStateMachine(final Set<ASMState> the_states,
+                              final Set<ASMEvent> the_events,
+                              final Set<ASMTransition> the_transition_function,
+                              final ASMState the_initial_state,
+                              final Set<ASMState> the_final_states) {
+    my_states = the_states;
+    my_events = the_events;
+    my_transition_function = the_transition_function;
+    my_initial_state = the_initial_state;
+    my_current_state = the_initial_state;
+    my_final_states = the_final_states;
+  }
+  
+  /**
+   * @return are we in the initial state?
+   */
+  public boolean initialState() {
```

Comment:
With these naming conventions, I'd expect `initialState()` and `finalState()` to return an `ASMState`, not a `boolean`.